src/dataset-builder/CohortEditor.ts

```diff
@@ -52,17 +52,28 @@ export const CriteriaView = (props: CriteriaViewProps) => {
 
   const [criteriaCount, setCriteriaCount] = useLoadedData<DatasetParticipantCountResponse>();
 
-  useEffect(() => {
-    setCriteriaCount(async () =>
-      DataRepo()
-        .dataset(datasetId)
-        .getCounts({
-          // Create a "cohort" to get the count of participants for this criteria on its own.
-          cohorts: [{ criteriaGroups: [{ criteria: [criteria], name: '', meetAll: true, mustMeet: true }], name: '' }],
-        })
-    );
-  }, [criteria, datasetId, setCriteriaCount]);
+  const debounceSetCriteriaCount = useRef(
```

Comment:
we want to use a useRef here because we only want one instance of the `debounce` function. If there are changes to any dependency inside of useRef (i.e datasetId, criteria) we can change it with `.current()`and passing in new values.